plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerLauncher.java

```diff
@@ -52,18 +55,47 @@
   private final Path launchScript;
   private ProcessorJsonRpcCommunication processorJsonRpcCommunication;
   private ExecuteClientCommandJsonRpcTransmitter executeCliendCommandTransmitter;
+  private EventService eventService;
+  private ProjectManager projectManager;
 
   @Inject
   public JavaLanguageServerLauncher(
       ProcessorJsonRpcCommunication processorJsonRpcCommunication,
-      ExecuteClientCommandJsonRpcTransmitter executeCliendCommandTransmitter) {
+      ExecuteClientCommandJsonRpcTransmitter executeCliendCommandTransmitter,
+      EventService eventService,
+      ProjectManager projectManager) {
     this.processorJsonRpcCommunication = processorJsonRpcCommunication;
     this.executeCliendCommandTransmitter = executeCliendCommandTransmitter;
+    this.eventService = eventService;
+    this.projectManager = projectManager;
     launchScript = Paths.get(System.getenv("HOME"), "che/ls-java/launch.sh");
   }
 
   public void sendStatusReport(StatusReport report) {
     LOG.info("{}: {}", report.getType(), report.getMessage());
+    if ("Started".equals(report.getType())) {
+      updateWorkspaceOnLSStarted();
+    }
+  }
+
+  private void updateWorkspaceOnLSStarted() {
+    projectManager
+        .getAll()
+        .forEach(
+            registeredProject -> {
+              if (!registeredProject.getProblems().isEmpty()) {
+                try {
+                  projectManager.update(registeredProject);
+                  eventService.publish(new ProjectUpdatedEvent(registeredProject.getPath()));
+                } catch (Exception e) {
```

Comment:
This would catch runtime exceptions, etc. I this what we want? I would rather use multi-catch to catch expected Exceptions.